arch/x86/adler32_avx512_vnni.c

```diff
@@ -0,0 +1,108 @@
+/* adler32_avx512_vnni.c -- compute the Adler-32 checksum of a data stream
+ * Based on Brian Bockelman's AVX2 version
+ * Copyright (C) 1995-2011 Mark Adler
+ * Authors:
+ *   Adam Stylinski <kungfujesus06@gmail.com>
+ *   Brian Bockelman <bockelman@gmail.com>
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include "../../zbuild.h"
+#include "../../zutil.h"
+
+#include "../../adler32_p.h"
+
+#include <immintrin.h>
+
+#if defined(X86_AVX512VNNI_ADLER32)
```

Comment:
Use `#ifdef` instead of `#if defined`. We usually use the shorter version if just evaluating one variable. Also it makes it same as adler32_avx512.c.